Develop

kdelange opened the pull request:
removed the key of dspaling

marieke-bijlsma commented on README.md:
```diff
@@ -1,22 +1,75 @@
+| /home/${home}/              | Shared       | Yes     | UIs, DAIs, SAIs, CNs | Only for personal preferences: small data == tiny quota.|
+| /groups/${group}/prm[0-9]/  | Shared       | Yes     | UIs, DAIs            | **p**e**rm**anent storage folders: for rawdata or *final* results that need to be stored for the mid/long term. |
+| /groups/${group}/tmp[0-9]/  | Shared       | No      | UIs, DAIs, CNs       | **t**e**mp**orary storage folders: for staged rawdata and intermediate results on compute nodes that only need to be stored for the short term. |
+| /groups/${group}/scr[0-9]/  | Local        | No      | Some UIs             | **scr**atch storage folders: same as **tmp**, but local sotrage as opposed to shared storage. Optional and available on all UIs. |
```
typo: "sotrage"

marieke-bijlsma commented on README.md:
```diff
@@ -1,22 +1,75 @@
+
+---
+
+### 2. Ansible playbooks openstack cluster
```
OpenStack instead of openstack?

pneerincx committed:
Added cargo (Rust's package manager and build tool) too.

Commit:
new user ohongnat

pneerincx committed:
Merge pull request #357 from BenjaminsM/develop

new user ohongnat

pneerincx committed:
Added marloes admin.

pneerincx committed:
Updated sudoers config for Gearshift.

Gerbenvandervries committed:
Merge pull request #358 from pneerincx/feature/new_admin

Added marloes admin.

marieke-bijlsma committed:
Merge pull request #356 from pneerincx/feature/add_libsodium

Added libsodium and accompanying libsodium-devel for cryptography.

marieke-bijlsma committed:
Merge pull request #359 from pneerincx/fix/sudoers

Updated sudoers config for Gearshift.

pneerincx committed:
Fixed code that checks if working rsync version is present and moved it to separate role to reduce redundancy.

erijpkema committed:
Merge pull request #361 from pneerincx/fix/check_rsync

Improved rsync handling

erijpkema committed:
Pin kernel by name.

We found that the ordering of the kernels is not the same across
instances.

erijpkema committed:
alerting updates

check for oom killer
added new hostnames for hypervisors
fixed alerts not showing hostname where alert was triggered (by adding
quotes)

pneerincx committed:
Merge pull request #362 from rug-cit-hpc/fix/alerts

alerting updates

pneerincx committed:
Use /bin/bash for shell tasks.

erijpkema committed:
Merge pull request #364 from pneerincx/fix/force_bash

Use /bin/bash for shell tasks.

pneerincx committed:
Updated circleci config to also lint deploy-os_servers.yml playbook and fixed a few linter errors in that playbook.

erijpkema committed:
Merge pull request #365 from pneerincx/lint_both_playbooks

Updated CI config

Commit:
new user mslofstra

pneerincx committed:
First configs files for Calculon (WIP).

marieke-bijlsma committed:
Merge pull request #367 from pneerincx/new_jumphosts

First configs files for Calculon (WIP).

pneerincx committed:
Merge pull request #366 from BenjaminsM/develop

new user mslofstra

marieke-bijlsma committed:
New vars for calculon

Gerbenvandervries committed:
Merge pull request #368 from marieke-bijlsma/develop

New vars for calculon

erijpkema committed:
Made the default kernel configurable.

pneerincx committed:
Updated CircleCI config for new ansible-linter 5.0.0, which requires explicit ansible version to install.

pneerincx committed:
Added explicit fetching of requirements from Ansible Galaxy.

erijpkema committed:
Merge pull request #369 from pneerincx/fix/updated_circleci_config

Updated CircleCI config for new ansible-linter 5.0.0

pneerincx committed:
Added ${HOME}/.ansible/roles to ANSIBLE_ROLES_PATH.

pneerincx committed:
Fixed yaml-lint errors.

pneerincx committed:
Fixed additional yaml-lint errors and replaced ansible-lint rule numbers (deprecated) with rule names.

pneerincx committed:
Use relative paths to resolve included tasks from another file.

pneerincx committed:
Updated parsing of ansible-lint results.

pneerincx committed:
Merge pull request #363 from rug-cit-hpc/fix/explicit-kernel-pinning

Pin kernel by name.

pneerincx committed:
Updated password for ssh_hosdt_signer CA key.

pneerincx committed:
Replaced "hosts: all" with host groups "jumphost + cluster + docs" te prevent including the "openstack_api" group containing localhost.

Gerbenvandervries committed:
Merge pull request #370 from pneerincx/fix/updated_circleci_config2

Updated CircleCI config for new ansible-lint 5.0.0

pneerincx committed:
Updated stdout_callback to use new Ansible 2.10.x syntax and switched from debug to yaml callback.

marieke-bijlsma committed:
Add openstack API to calculon_hosts

marieke-bijlsma committed:
Merge pull request #371 from pneerincx/small_fixes

Small fixes

pneerincx committed:
Merge pull request #372 from marieke-bijlsma/develop

Add openstack API to calculon_hosts

pneerincx committed:
Temporarily added tag to skip task suffering from know bug. Can be removed again once the bug is fixed.

pneerincx committed:
Added floating IP of new lobby jumphost to group_vars for Calculon.

marieke-bijlsma committed:
Merge pull request #373 from pneerincx/develop

Updates for new lobby jumphost

Gerbenvandervries committed:
new userkey

pneerincx committed:
Merge pull request #374 from Gerbenvandervries/develop

new userkey

marieke-bijlsma committed:
ip_addresses for calculon cluster

Gerbenvandervries committed:
Merge pull request #375 from marieke-bijlsma/develop

ip_addresses for calculon cluster

marieke-bijlsma committed:
Add calculon management and storage ip addresses

Gerbenvandervries committed:
Merge pull request #376 from marieke-bijlsma/develop

Add calculon management and storage ip addresses

erijpkema committed:
A single role playbook for datahandling.

mheijkoop committed:
Merge pull request #377 from rug-cit-hpc/fix/single-role-playbook-datahandling

A single role playbook for datahandling.

pneerincx committed:
Added extra rsc[0-9] lustre mounts from data handling storage to NHC configs.

marieke-bijlsma committed:
Add additional_etc_hosts to calculon vars

erijpkema committed:
Merge pull request #378 from pneerincx/fix/add_rsc_to_nhc_configs

Added extra rsc[0-9] lustre mounts from data handling storage to NHC configs.

pneerincx committed:
Merge pull request #379 from marieke-bijlsma/develop

Add additional_etc_hosts to calculon vars

pneerincx committed:
Simplified and improved regexes to find subgroups.

pneerincx committed:
WIP

pneerincx committed:
WIP

marieke-bijlsma committed:
Merge pull request #380 from pneerincx/fix/subgroup_regex

Simplified and improved regexes to find subgroups.

pneerincx committed:
Updated lustre mount options per advise from Robin.

pneerincx committed:
Updated lustre mount options per advise from Robin.

pneerincx committed:
Finished new lustre_client role, which deploy the latest version available and relies on DKMS.

pneerincx committed:
Finished new lustre_client role, which deploy the latest version available and relies on DKMS.

pneerincx committed:
Updated group_vars for Gearshift and Nibbler to use new lustre_client role. Version of Lustre and Kernel is not defined in this repo: we rely on other software (Spacewalk/Pulp) to pin Lustre and Kernel versions.

pneerincx committed:
Updated group_vars for Gearshift and Nibbler to use new lustre_client role. Version of Lustre and Kernel is not defined in this repo: we rely on other software (Spacewalk/Pulp) to pin Lustre and Kernel versions.

pneerincx committed:
Updated synchronize to ansible.posix.synchronize, which includes two bug fixes we now no longer need to patch after each Ansible update.

pneerincx committed:
Updated README: removed section on how to patch bugs, which are now fixed.

pneerincx committed:
Fixed typos in comments.

pneerincx committed:
Fixed typos in comments.

pneerincx committed:
Added "-o ldif-wrap=no" option to ldapsearch commands to prevent line wrapping and make it easier to parse the query results.

pneerincx committed:
Added "-o ldif-wrap=no" option to ldapsearch commands to prevent line wrapping and make it easier to parse the query results.

pneerincx committed:
Added security rule to allow inbound LDAPS traffic.

pneerincx committed:
Added security rule to allow inbound LDAPS traffic.

pneerincx committed:
Added Pulp installer Ansible collection to requirements.

pneerincx committed:
Added repo server group to various templates, roles, playbooks, etc.

pneerincx committed:
Added repo server group to various templates, roles, playbooks, etc.

pneerincx committed:
Added repo server group to various templates, roles, playbooks, etc.

pneerincx committed:
Added repo server for Nibbler cluster.

pneerincx committed:
Switched role order in cluster.yml, because logins rele depends on RPM from EPEL repo.

pneerincx committed:
Switched role order in pulp_server.yml, because logins rele depends on RPM from EPEL repo.

pneerincx committed:
Switched role order in pulp_server.yml, because logins rele depends on RPM from EPEL repo.

pneerincx committed:
Make sure repo server is up to date.

marieke-bijlsma committed:
Merge pull request #381 from pneerincx/fix/synchronize

Fix/synchronize

pneerincx committed:
Added Ger and updated his public key.

marieke-bijlsma committed:
Merge pull request #382 from pneerincx/feature/admin_ger

Added Ger and updated his public key.

pneerincx committed:
Created new role to apply updates with handler to reboot for kernel updates and to create a cronjob to reboot for kernel updates on machines that receive dayly updates.

Commit:
added new user dwilliam and updated keys of user ebenetti

pneerincx committed:
Merge pull request #383 from BenjaminsM/develop

added new user dwilliam and updated keys of user ebenetti

pneerincx committed:
Make sure the needs-restarting utility is installed.

pneerincx committed:
Updated public key for mswertz.

marieke-bijlsma committed:
Merge pull request #384 from pneerincx/fix/updated_key

Updated public key for mswertz.

pneerincx committed:
Fixed documentation for local scratch space on clusters with nodes that do not have any local scratch space.

pneerincx committed:
Removed requirement to use chmod in AppleScript app, which prevents issues with "app translocation" security feature from macOS, which executes the app from a read-only, temporary partition/volume.

marieke-bijlsma committed:
Merge pull request #385 from pneerincx/fix/documentation

Fix/documentation

pneerincx committed:
Nasty merge conflict mess resolved.

Commit:
new user mbeukers

pneerincx committed:
Merge pull request #386 from BenjaminsM/develop

new user mbeukers

pneerincx committed:
Fixed typo.

pneerincx committed:
Re-added pulp_installer role that was lost in fixing version conflicts.

marieke-bijlsma committed:
Merge pull request #387 from pneerincx/fix/docs

Fixed SSH key type in instructions.

pneerincx committed:
Added pulp client role (WIP).

marieke-bijlsma committed:
Add installation of icommands

pneerincx committed:
Merge pull request #388 from marieke-bijlsma/develop

pneerincx committed:
Finished playbook to deploy a Pulp server. Switched Nibbler from Spacewalk to Pulp and introduced variables to keep other clusters on Spacewalk for now.

Commit:
new user rsietsma

pneerincx committed:
Merge pull request #390 from BenjaminsM/develop

new user rsietsma

marieke-bijlsma committed:
Small changes to connect to ldap proxy

Gerbenvandervries committed:
Merge pull request #389 from pneerincx/feature/pulp

Feature/pulp

pneerincx committed:
Merge branch 'develop' into develop

pneerincx committed:
Merge pull request #391 from marieke-bijlsma/develop

Small changes to connect to ldap proxy

pneerincx committed:
Merge branch 'fix/ldap' of https://github.com/pneerincx/league-of-robots into fix/ldap
Regular pull.

marieke-bijlsma committed:
Merge pull request #392 from pneerincx/fix/ldap

Fix/ldap

pneerincx committed:
Added iRODS Pulp repo.

pneerincx committed:
Flush handlers to update ca-trust before quering the HTTPS URLs that rely on those CA certificates. Added task to remove RHN/SpaceWalk code/configs/packages as those will conflict with our new Pulp repos.

erijpkema committed:
Added fokke and bob to the admins.

erijpkema committed:
Vim's yaml plugin failed me.

It inserted an identation where there should be none.

pneerincx committed:
Merge pull request #393 from rug-cit-hpc/Add_fokke

Added fokke and bob to the admins.

Commit:
added new users fei gao and david ruvolo

pneerincx committed:
Merge pull request #394 from BenjaminsM/develop

added new users fei gao and david ruvolo

pneerincx committed:
Updated galaxy-requirements.yml, because we need a newer version of Pulp with bugfix for repos that failed to sync with their remotes.

pneerincx committed:
Added squeezer collection for managing pulp repos.

pneerincx committed:
Fixed typo in comment.

pneerincx committed:
Added pulp_server role (WIP).

pneerincx committed:
Added documentation for working with Pulp (WIP).

pneerincx committed:
Added repoadmin user.

pneerincx committed:
Disable PID for all Slurm daemons and let systemd automagically detect the PID of any running Slurm daemons.

marieke-bijlsma committed:
Merge pull request #395 from pneerincx/feature/new_admin_user

Added repoadmin user.

marieke-bijlsma committed:
Merge pull request #396 from pneerincx/fix/slurm_systemd_service_files

Disable PID files for all Slurm daemons.

pneerincx committed:
WIP.

pneerincx committed:
Fixed merge conflict and moved repoadmin user to pulp_server role defaults.

pneerincx committed:
Added new single role playbook for pulp_server role.

pneerincx committed:
Bugfixes.

pneerincx committed:
Bugfix: create missing folders.

pneerincx committed:
Additional improvements for pulp_server role.

pneerincx committed:
Added documentation for configuring Pulp repo management servers.

pneerincx committed:
Added pulp password and secret to secrets.yml template and secrets.yml for Nibbler.

pneerincx committed:
Fixed background

pneerincx committed:
Use single quotes to ensure literal strings in *.toml config file.

pneerincx committed:
Renamed variables for pulp_server role to prevent clash with vars from the pulp-intaller.

pneerincx committed:
Refer to documentation.

pneerincx committed:
Integrated Pulp stuff in main cluster.yml playbook and removed separate deploy-pulp_server.yml playbook used for development.

pneerincx committed:
Small bug fixes for pulp_client role.

pneerincx committed:
Split main cluster.yml playbook in several chunks per inventory group, so it is easier to to (re)deploy all roles for an inventory group.

pneerincx committed:
Bugfixes for documentation/Configuring_Pulp.md

pneerincx committed:
Added section to explain job priority and backfill.

pneerincx committed:
Fixed color.

pneerincx committed:
Merge branch 'fix/lustre' of https://github.com/pneerincx/league-of-robots into fix/lustre
Regular pull.

pneerincx committed:
Added repo for Lustre client and updated documentation accordingly.

marieke-bijlsma committed:
Merge pull request #399 from pneerincx/feature/improved_online_docs

Feature/improved online docs

pneerincx committed:
Fixed typo in documentation.

erijpkema committed:
Merge pull request #398 from pneerincx/feature/pulp/2

Additional work for pulp_server role

pneerincx committed:
Fixed Slurm fair share configuration.

pneerincx committed:
Fixed Lua comments.

kdelange committed:
added keys to user

pneerincx committed:
Merge pull request #402 from kdelange/develop

added keys to user

pneerincx committed:
Fixed indentation in users.yml

marieke-bijlsma committed:
Merge pull request #401 from pneerincx/fix/slurm_fair_share

Fixed Slurm fair share configuration.

marieke-bijlsma committed:
Merge pull request #403 from pneerincx/develop

Fixed indentation in users.yml

erijpkema committed:
Merge pull request #400 from pneerincx/fix/lustre

Fix/lustre

pneerincx committed:
Added new user Sandi.

scimerman committed:
Merge pull request #404 from pneerincx/develop

Added new user Sandi.

pneerincx committed:
Added sandi admin to Talos, Hyperchicken and Nibbler.

scimerman committed:
Merge pull request #405 from pneerincx/develop

Added sandi admin to Talos, Hyperchicken and Nibbler.

pneerincx committed:
Added logrotate role.

pneerincx committed:
Create prm, tmp and rsc mounts for umcg-sysops group, so they can test cluster functionality, without requiring access to data from other groups.

erijpkema committed:
Merge pull request #407 from pneerincx/feature/shared_stroge_mounts_for_sysops

Create prm, tmp and rsc mounts for umcg-sysops group

scimerman committed:
Updated the ldap task call to be compatible with newer standards

pneerincx committed:
Merge pull request #408 from scimerman/develop

Updated the ldap task call to be compatible with newer standards.

scimerman committed:
Merge pull request #406 from pneerincx/feature/logrotate

Added logrotate role.

pneerincx committed:
Added new user.

scimerman committed:
Merge pull request #409 from pneerincx/feature/added_regular_user

Added new user.

pneerincx committed:
Fixes for new ansible-linter issues and fixed bug in CircleCI config.

pneerincx committed:
Bugfix.

pneerincx committed:
Fixed yaml linter issues.

pneerincx committed:
Fixed yaml linter issues.

pneerincx committed:
Minor improvements.

pneerincx committed:
Fixed situation where grep would fail when there are no erros.

pneerincx committed:
Added options to disable colored output, which would break parsing the result with sed.

pneerincx committed:
Merge pull request #410 from pneerincx/fix/ci

Fixes for new ansible-linter issues and fixed bug in CircleCI config.

pneerincx committed:
Improved QoS settings (WIP).

pneerincx committed:
Improved QoS settings (WIP).

pneerincx committed:
First commits for migrating Hyperchicken cluster to new Embassy cloud (Work in Progress).

scimerman committed:
Merge pull request #411 from pneerincx/feature/new_hyperchicken

First commits for migrating Hyperchicken cluster to new Embassy cloud.

pneerincx committed:
Merge branch 'feature/slurm_qos_interactive' of https://github.com/pneerincx/league-of-robots into feature/slurm_qos_interactive
Regular pull.

scimerman committed:
Minor changes to group_vars/template/ip_addresses.yml.j2 and redeployed Hyperchicken new IPs

pneerincx committed:
Merge pull request #412 from scimerman/develop

Minor changes to group_vars/template/ip_addresses.yml.j2 and redeploy…

pneerincx committed:
Updated Gearshift group_vars for new umcg-hematology group.

scimerman committed:
Merge pull request #413 from pneerincx/feature/hematology

Updated Gearshift group_vars for new umcg-hematology group.

pneerincx committed:
Switched ssh-ldap-wrapper back to Bash version to remove problematic Python dependencies.

pneerincx committed:
Bugfix.

pneerincx committed:
Bugfix.

pneerincx committed:
One more bugfix.

pneerincx committed:
Disable LDAP lookups for local accounts and groups. Prevents delays for local admin users when LDAP is down.

pneerincx committed:
Bugfix.

marieke-bijlsma committed:
Merge pull request #414 from pneerincx/feaure/simplified_ldap_wrapper

Simplified LDAP wrapper

pneerincx committed:
Replaced key for one user.

scimerman committed:
Merge pull request #415 from pneerincx/fix/publickey

Replaced key for one user.

scimerman committed:
Local admin centos removed, removed spacewalk yum_repos from jumphost and docs

pneerincx committed:
Updating documentation (WIP).

pneerincx committed:
Merge pull request #416 from scimerman/develop

Local admin centos removed, removed spacewalk yum_repos from jumphost…

pneerincx committed:
Fixed bug where Ansible would hang waiting for su to return, which would never happen.

scimerman committed:
Merge pull request #417 from pneerincx/fix/initialize_home_dirs

Fix for correct initialization of home dirs.

pneerincx committed:
Updated instructions for authentication to OpenStack API using Application Credentials.

scimerman committed:
Merge pull request #418 from pneerincx/fix/openstack_cli_authentication

Updated instructions for authentication to OpenStack API using Application Credentials.

pneerincx committed:
Updating documentation (WIP).

pneerincx committed:
Tweaked SchedulerParameters in slurm.conf.

pneerincx committed:
Switched Nibbler to new and from Merlin accessible LDAP.

pneerincx committed:
Removed unused ldap_port variable and removed ldaps_port variable: always use default port 636 for LDAP.

scimerman committed:
Merge pull request #419 from pneerincx/fix/nibbler_ldap_client

Switched Nibbler to new and from Merlin accessible LDAP.

marieke-bijlsma committed:
Updated variables and added ldaps uri

marieke-bijlsma committed:
solve merge conflict

pneerincx committed:
Merge pull request #420 from marieke-bijlsma/develop

Updated variables and added ldaps uri

pneerincx committed:
Added autmatic fetching of galaxy-requirements.yml the hacky way. Does not get any better with current limited ansible-galaxy dependency management support.

pneerincx committed:
Bugfix for wrong path.

pneerincx committed:
Fixed linter issue.

erijpkema committed:
Merge pull request #421 from pneerincx/fix/deps

Fix/deps

pneerincx committed:
Updated online documentation for new QoS level "interactive".

pneerincx committed:
Updated online documentation for new QoS level "interactive".

pneerincx committed:
Merge branch 'develop' into feature/slurm_qos_interactive

pneerincx committed:
Fixed formatting in docs.

pneerincx committed:
Merge branch 'feature/slurm_qos_interactive' of https://github.com/pneerincx/league-of-robots into feature/slurm_qos_interactive
Regular pull.

pneerincx committed:
Fixed formatting in docs.

pneerincx committed:
Removed preempt from main/routing QoS "priority".

scimerman committed:
Merge pull request #422 from pneerincx/feature/slurm_qos_interactive

Feature: Slurm QoQ interactive

pneerincx committed:
Fixed Slurm prolog, where $SLURM_JOB_QOS does not exist.

pneerincx committed:
Fixed config errors for Talos that would result in NHC taking nodes offline.

erijpkema committed:
Merge pull request #424 from pneerincx/fix/slurm_prolog_failure

Slurm configuration fixes.

marieke-bijlsma committed:
Removed boxy and umcg from calculon vars and added both gattaca machines

marieke-bijlsma committed:
Fix location of gattaca machines in variable files

pneerincx committed:
Merge pull request #426 from marieke-bijlsma/develop

Removed boxy and umcg from calculon vars and added both gattaca machines

kdelange committed:
new user and small bug fix

kdelange committed:
fix bugfix

pneerincx committed:
Merge pull request #427 from kdelange/develop

new user and small bug fix

pneerincx committed:
Updates for new version of mkdocs (WIP).

pneerincx committed:
Fixed broken hyperlinks in online docs and document repo manager based on variable now that we use either Spacewalk or Pulp.

pneerincx committed:
Fixed yaml lint error.

scimerman committed:
Merge pull request #428 from pneerincx/fix/mkdocs

Fix/mkdocs

scimerman committed:
Pulp server version locked: pulp_installer==3.12.2 pulpcore==3.12.2 pulp-rpm==3.10.0

scimerman committed:
Updated the documentation about the pulp bug

scimerman committed:
Typo resolved

scimerman committed:
Fixed yaml typo

pneerincx committed:
Merge pull request #432 from scimerman/develop

Pulp version locking of pulp_installer, pulpcore and pulp-rpm. Working solution for now

pneerincx committed:
Include external IP addresses in /etc/hosts of new jumhost instead of internal addresses as new jumphost is hosted on a different site.

pneerincx committed:
Added missing firewall role.

pneerincx committed:
Remove unused redundant copy of geerlingguy.firewall from roles subdir. We use a copy from Ansible Galaxy instead and this is installed outside this repo.

pneerincx committed:
Pion version numbers for roles imported from Ansible Galaxy to prevent surprises.

pneerincx committed:
Pin version numbers for roles imported from Ansible Galaxy to prevent surprises.

pneerincx committed:
Merge branch 'fix/new_lobby' of https://github.com/pneerincx/league-of-robots into fix/new_lobby
Regular pull.

scimerman committed:
Merge pull request #433 from pneerincx/fix/new_lobby

Fix/new lobby

kdelange committed:
removed user
